src/mesh/domain.hpp

```diff
@@ -140,116 +140,131 @@ class IndexShape {
     }
   }
 
-  KOKKOS_INLINE_FUNCTION const IndexRange
-  GetBoundsI(const IndexDomain &domain) const noexcept {
-    return (domain == IndexDomain::interior) ? x_[0] : IndexRange{is(domain), ie(domain)};
+  KOKKOS_INLINE_FUNCTION const IndexRange GetBoundsI(const IndexDomain &domain,
+                                                     TE el = TE::CC) const noexcept {
+    return (domain == IndexDomain::interior && el == TE::CC)
+               ? x_[0]
+               : IndexRange{is(domain, el), ie(domain, el)};
   }
```

Comment:
probably important to have this functionality, but FWIW I think it's probably going to be pretty common for people to use the cell centered domains and add the offsets in by hand

Reply:
One can do that, having this option certainly simplifies writing more generic code in my experience. In my opinion, I think we should try to enforce explicitly passing the topological type to `GetBounds*` where practical since it results in code that is easier to understand.